src/table/__tests__/resizable-columns.test.tsx

```diff
@@ -431,3 +438,37 @@ describe('column header content', () => {
     expect(getResizeHandle(1)).toHaveAttribute('aria-roledescription', 'resize button');
   });
 });
+
+test('should set last column width to "auto" when container width exceeds total column width', () => {
+  const totalColumnsWidth = 150 + 300;
+
+  let outsideCb: (entry: ContainerQueryEntry) => void = () => {};
+  jest.mocked(useResizeObserver).mockImplementation((_target, cb) => {
+    // The table uses more than one resize observer.
+    // The callback must be triggered for all to ensure the expected one is targeted as well.
```

Comment:
The would have been nice, but the target is the same as currently we attach multiple observers to the same wrapper measure DOM node. This can be fixed after 1743 and 1745 are merged by ensuring a single observer with some API for registering listeners.

Reply:
Sounds like a use case for a map:

```js
const observedNodes = new Map<HTMLElement, Array<Callback>>();

// ....

useLayoutEffect(() => {
  cb({ contentBoxWidth: initialValue } as unknown as ContainerQueryEntry);

  observedNodes.get(target).push(cb);
  return () => {
     observedNodes.set(target, observedNodes.get(target)).filter(value => cb !== value);      
  }
}, []);
```

usage in tests

```
function notifyResize(target, newSize) {
    act(() => {
       observedNodes.get(target).forEach(cb => cb(newSize))
    })
}
```

Why did I put `act`? See another related thread: https://github.com/cloudscape-design/components/pull/1727#discussion_r1387867723

Reply:
Makes sense, I will add the `act`. See no big reason to have that observed nodes map though. W/o a map all table resize observer callbacks are triggered but I don't expect those to have any effect on the feature anyways.

Reply:
Still not fixed. If you do not want a `Map`, you could at least have an array of callbacks, to avoid hard-coded `prev` and `cb` recursive calls